internal/actions/log.go

```diff
@@ -0,0 +1,112 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/speakeasy-api/sdk-generation-action/internal/environment"
+	"gopkg.in/yaml.v3"
+)
+
+const defaultAPIURL = "https://api.prod.speakeasyapi.dev"
+
+type logProxyLevel string
+
+const (
+	logProxyLevelInfo  logProxyLevel = "info"
+	logProxyLevelError logProxyLevel = "error"
+)
+
+type logProxyEntry struct {
+	LogLevel logProxyLevel          `json:"log_level"`
+	Message  string                 `json:"message"`
+	Source   string                 `json:"source"`
+	Tags     map[string]interface{} `json:"tags"`
+}
+
+func LogActionResult() error {
+	key := os.Getenv("SPEAKEASY_API_KEY")
+	if key == "" {
+		return fmt.Errorf("no speakeasy api key available.")
+	}
+
+	logLevel := logProxyLevelInfo
+	logMessage := "Success in Github Action"
+	if !strings.Contains(strings.ToLower(os.Getenv("GH_ACTION_RESULT")), "success") {
+		logLevel = logProxyLevelError
+		logMessage = "Failure in Github Action"
+	}
+
+	request := logProxyEntry{
+		LogLevel: logLevel,
+		Message:  logMessage,
+		Source:   "gh_action",
+		Tags: map[string]interface{}{
+			"target":            os.Getenv("TARGET"),
+			"gh_repository":     fmt.Sprintf("https://github.com/%s", os.Getenv("GITHUB_REPOSITORY")),
+			"gh_action_version": os.Getenv("GH_ACTION_VERSION"),
+			"gh_action_step":    os.Getenv("GH_ACTION_STEP"),
+			"gh_action_result":  os.Getenv("GH_ACTION_RESULT"),
+			"gh_action_run":     fmt.Sprintf("https://github.com/%s/actions/runs/%s", os.Getenv("GITHUB_REPOSITORY"), os.Getenv("GITHUB_RUN_ID")),
+			"run_origin":        "gh_action",
+		},
+	}
+
+	languages := environment.GetLanguages()
+	languages = strings.ReplaceAll(languages, "\\n", "\n")
+	langs := []string{}
+	if err := yaml.Unmarshal([]byte(languages), &langs); err != nil {
+		fmt.Println("No language provided in github actions config.")
+	}
+	if len(langs) > 0 {
+		request.Tags["language"] = langs[0]
```

Comment:
Seems like it would be pretty relevant to collect all languages?

Reply:
We can do that in a separate tag. The GH action accepts multiple inputs of languages for generation but only uses one really.